adit/dicom_web/utils/wadors_utils.py

```
@@ -40,58 +58,227 @@ def callback(ds: Dataset) -> None:
-            while True:
-                queue_ds = await queue.get()
-                if queue_ds is None:
-                    break
-                yield queue_ds
+        await fetch_task
 
     except ExceptionGroup as eg:
-        # Extract the first relevant exception
         for exc in eg.exceptions:
             if isinstance(exc, RetriableDicomError):
                 raise ServiceUnavailableApiError(str(exc))
```

Ruff (0.15.0)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)